pkg/secrets/push_to_file/secrets_groups.go

```diff
@@ -0,0 +1,235 @@
+	}
+
+	insertSecret := func(name string, secretPath string) error {
+		secretPath = path.Clean(secretPath)
+		if secretPath == "." {
+			return log.RecordedError(messages.CSPFK051E, "Invalid secret path", groupName)
+		}
+		secrets[name] = secretsPathsPrefix + secretPath
+
+		return nil
+	}
+
+	for _, secret := range unpacked {
+		switch val := secret.(type) {
+		case string:
+			{
+				name := val[strings.LastIndex(val, "/")+1:]
+				if name == "" {
+					return nil, log.RecordedError(messages.CSPFK051E, "Invalid secret name", groupName)
+				}
+
+				if err := insertSecret(name, val); err != nil {
+					return nil, err
+				}
+			}
+
+		case map[string]interface{}:
+			{
+				for alias, secretPath := range val {
+					secretPath, _ := secretPath.(string)
+					if err := insertSecret(alias, secretPath); err != nil {
+						return nil, err
+					}
+				}
+			}
+
+		default:
+			return nil, log.RecordedError(messages.CSPFK051E, "Unknown secret format", groupName)
+		}
+	}
+
+	if len(secrets) == 0 {
+		return nil, log.RecordedError(messages.CSPFK051E, "No valid secrets found", groupName)
+	}
+
+	return secrets, nil
+}
+
+func parseSecretsPathPrefix(groupName string, secretsPathPrefix string) (string, error) {
+	secretsPathPrefix = path.Clean("/" + secretsPathPrefix)
+	if secretsPathPrefix == "." || secretsPathPrefix == "/" {
+		return "/", nil
+	}
+
+	return secretsPathPrefix + "/", nil
+}
+
+func parseFilePath(groupName string, filePath string) (string, error) {
+	if strings.HasPrefix(filePath, "/") {
+		return "", log.RecordedError(messages.CSPFK052E, fmt.Sprintf("%s%s", SECRETS_GROUP_FILE_PATH_PREFIX, groupName))
+	}
+
+	if strings.HasSuffix(filePath, "/") {
+		filePath = path.Clean(filePath) + "/"
+	} else {
+		filePath = path.Clean(filePath)
+	}
+
+	if filePath == "." {
+		return DEFAULT_FILE_PATH, nil
+	}
+
+	return DEFAULT_FILE_PATH + filePath, nil
+}
+
+func parseFileType(groupName string, fileType string) (SecretsFileType, error) {
+	switch strings.ToLower(fileType) {
+	case "":
+		return FILE_TYPE_YAML, nil
+	case "yaml":
+		return FILE_TYPE_YAML, nil
+	case "json":
+		return FILE_TYPE_JSON, nil
+	case "dotenv":
+		return FILE_TYPE_DOTENV, nil
+	case "bash":
+		return FILE_TYPE_BASH, nil
+	case "plaintext":
+		return FILE_TYPE_PLAINTEXT, nil
+	default:
+		return FILE_TYPE_NONE, log.RecordedError(messages.CSPFK053E, fmt.Sprintf("%s%s", SECRETS_GROUP_FILE_TYPE_PREFIX, groupName), fileType)
+	}
+}
+
+func parseFileTemplate(groupName string, fileTemplate string) (*template.Template, error) {
+	if fileTemplate == "" {
+		return nil, nil
+	}
+
+	t, err := template.New("FileTemplate").Parse(fileTemplate)
+	if err != nil {
+		return nil, log.RecordedError(messages.CSPFK054E, fmt.Sprintf("%s%s", SECRETS_GROUP_FILE_TEMPLATE_PREFIX, groupName), err.Error())
+	}
+
+	return t, nil
+}
+
+func validateGroupInfo(groupName string, groupInfo SecretsGroupInfo) error {
+	if groupInfo.FileTemplate != nil {
+		if strings.HasSuffix(groupInfo.FilePath, "/") {
```

Comment:
I think we need a check for the `FilePath` being unspecified (i.e. a null string) (in addition to the check for trailing `/`)?

Reply:
There's no null string just `""`